### PR TITLE
Close open file handles immediately when ejecting track

### PR DESCRIPTION
**Pull request by uklotzde**
This issue has been reported in the forums:
http://www.mixxx.org/forums/viewtopic.php?f=3&t=8848

### CODE REVIEW

File: src/engine/enginebuffer.cpp [hunk 1]
```diff
@@ -562,6 +562,8 @@ void EngineBuffer::ejectTrack() {
     doSeekFractional(0.0, SEEK_EXACT);
     m_pause.unlock();
 
+    m_pReader->newTrack(m_pCurrentTrack);
```
Comment (rryan) [inline src/engine/enginebuffer.cpp]: ```TrackPointer()``` might be more readable. 

Could you add a comment that it's to clean up state in CachingReader?

File: src/engine/cachingreaderworker.cpp [hunk 2]
```diff
@@ -75,12 +77,13 @@ void CachingReaderWorker::run() {
 
     Event::start(m_tag);
     while (!load_atomic(m_stop)) {
-        if (m_newTrack) {
+        if (m_newTrackAvailable) {
```
Comment (rryan) [inline src/engine/cachingreaderworker.cpp]: Was the pointer not a good enough boolean? Is this for readability?

**Comment by uklotzde** [comment]
Now CachingReaderWorker needs to wake up even if m_pNewTrack == nullptr ;)

**Comment by rryan** [comment]
> Now CachingReaderWorker needs to wake up even if m_pNewTrack == nullptr ;)

Oh duh, :man_facepalming: thanks

**Comment by uklotzde** [comment]
Don't worry. I made the same mistake and wondered, why I unloading didn't 
work as expected :D

**Comment by uklotzde** [comment]
Option would be a perfect match here ;)

**Comment by JosepMaJAZ** [comment]
I confirm that it works as expected (file becomes editable with a 3rd party program once the track is ejected).
No side effects observed.

**Pull request merged**